library/src/main/java/com/jwoolston/android/uvc/streaming/SourceClockReference.kt

```diff
@@ -0,0 +1,25 @@
+package com.jwoolston.android.uvc.streaming
+
+import java.nio.ByteBuffer
+
+/**
+ * @author Jared Woolston (Jared.Woolston@gmail.com)
+ */
+class SourceClockReference internal constructor(data: ByteBuffer) {
```

Comment:
Since this class is immutable you could instead just use a data class then you don't need to implement `toString` as kotlin will generate it for you as well as a sane hashCode and equality check.

```kotlin
data class SourceClockReference (
   val sourceTime: Int,
   val tokenCounter: Int
)
```

Obviously, this is missing the extraction from the buffer but you can solve that by either adding a companion function for creating the class from the ByteBuffer or what I would do is write an extension method.

```kotlin
val ByteBuffer.asSourceClockReference: SourceClockReference
   get() = SourceClockReference(tokenCounter = data.int, sourceTime = data.int)
```

then simply

```koltin
val sourceClockReference = byteBuffer.asSourceClockReference
```